internal/batches/executor/executor.go

```diff
@@ -62,6 +63,7 @@ type newExecutorOpts struct {
 	Parallelism          int
 	Timeout              time.Duration
 	TempDir              string
+	AllowPathMounts      bool
 	WriteStepCacheResult func(ctx context.Context, stepResult execution.AfterStepResult, task *Task) error
```

Comment:
I think this one you also want to restrict when mounts are used. Otherwise, step-wise caching wouldn't trigger on changed files either.